packages/jaeger-ui/src/model/ddg/transformDdgData.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  PathElem,
+  TDdgModel,
+  TDdgPayload,
+  TDdgPayloadEntry,
+  TDdgPath,
+  TDdgPathElemsByDistance,
+  TDdgServiceMap,
+  TDdgVisibilityIdxToPathElem,
+} from './types';
+
+const stringifyPayloadEntry = ({ service, operation }: TDdgPayloadEntry) => `${service}::${operation}`;
```

Comment:
For lexicographical comparison, `:` is before some alphanumerics and after others:

- `a:op` is after `a0::op`
- `a:op` is before `aa::op`

I suggest using `\v` as the delimiter as it's very unlikely to be present in a service or operation name and is consistent in its lexicographical sorting for alphanumerics.

- `a\vop` is before `a0::op`
- `a\vop` is before `aa::op`

Reply:
changed to `\v` so it's easier to reason about (which is important) but ultimately it doesn't affect the outcome since order rationale doesn't matter, just order consistency.